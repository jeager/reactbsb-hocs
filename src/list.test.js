import React from 'react'
import {render, waitForElement, act} from 'react-testing-library'
// this adds custom jest matchers from jest-dom
import 'jest-dom/extend-expect'
import List from './list'


it('fetches data from server', async () => {
  const mockSuccess = [
    {
      "name": "João",
      "lastname": "Ninguém"
    }, {
      "name": "Jorge",
      "lastname": "Alguém"
    }
  ];
  
  const mockJsonPromise = Promise.resolve(mockSuccess);
  const mockFetchPromise = Promise.resolve({
    json: () => mockJsonPromise,
  })
  jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise)

  act(()=> {

  })
  const listComponent = render(<List/>)

  expect(global.fetch).toHaveBeenCalledTimes(1);
  expect(global.fetch).toHaveBeenCalledWith('http://localhost:3001/person-data');

  // https://github.com/facebook/react/issues/14769
  const renderedList = await waitForElement(() => listComponent.getByTestId('people-list'))

  expect(renderedList).toMatchSnapshot()
})