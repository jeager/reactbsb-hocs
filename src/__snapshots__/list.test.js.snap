// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches data from server 1`] = `
<ul
  data-testid="people-list"
>
  <li>
    João Ninguém
  </li>
  <li>
    Jorge Alguém
  </li>
</ul>
`;
